### PR TITLE
Implement TransactionalSession as part of the persistence

### PR DESCRIPTION
**Pull request by tmasternak**

### CODE REVIEW

File: src/NServiceBus.NHibernate.TransactionalSession.AcceptanceTests/EndpointConfigurer.cs [hunk 1]
```diff
@@ -0,0 +1,25 @@
+﻿using System;
+using System.Threading.Tasks;
+using NServiceBus;
+using NServiceBus.AcceptanceTesting.Support;
+
+public abstract class EndpointConfigurer : IConfigureEndpointTestExecution
```
Comment (danielmarbach) [inline src/NServiceBus.NHibernate.TransactionalSession.AcceptanceTests/EndpointConfigurer.cs:6]: Forgive my ignorance. Why was this abstraction needed?

File: src/NServiceBus.NHibernate.TransactionalSession/NHibernateOpenSessionOptions.cs [hunk 2]
```diff
@@ -0,0 +1,16 @@
+﻿namespace NServiceBus.TransactionalSession;
```
Comment (danielmarbach) [inline src/NServiceBus.NHibernate.TransactionalSession/NHibernateOpenSessionOptions.cs:1]: Nit: Can we use the curly namespaces here to align with the rest?

File: src/NServiceBus.NHibernate.TransactionalSession/NHibernateTransactionalSession.cs [hunk 3]
```diff
@@ -0,0 +1,25 @@
+﻿namespace NServiceBus.TransactionalSession;
```
Comment (danielmarbach) [inline src/NServiceBus.NHibernate.TransactionalSession/NHibernateTransactionalSession.cs:1]: Nit: Can we use the curly namespaces here to align with the rest?

File: src/NServiceBus.NHibernate.TransactionalSession/NHibernateTransactionalSessionOptionsCustomization.cs [hunk 4]
```diff
@@ -0,0 +1,16 @@
+﻿namespace NServiceBus.TransactionalSession;
```
Comment (danielmarbach) [inline src/NServiceBus.NHibernate.TransactionalSession/NHibernateTransactionalSessionOptionsCustomization.cs:1]: Nit: Can we use the curly namespaces here to align with the rest?

File: src/NServiceBus.NHibernate.TransactionalSession/NHibernateTransactionalSessionOptionsCustomization.cs [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+
+class NHibernateTransactionalSessionOptionsCustomization : IOpenSessionOptionsCustomization
+{
+    readonly string endpointName;
+
+    public NHibernateTransactionalSessionOptionsCustomization(string endpointName) => this.endpointName = endpointName;
```
Comment (danielmarbach) [inline src/NServiceBus.NHibernate.TransactionalSession/NHibernateTransactionalSessionOptionsCustomization.cs:7]: Technically you could also inject readonly settings and then from there get the endpoint name. Both approaches are valid. Just wanted to mention it

**Review by danielmarbach — APPROVED** [review]

File: src/NServiceBus.NHibernate.TransactionalSession/NHibernateTransactionalSessionOptionsCustomization.cs (hunk 5 above)
Comment (tmasternak) [reply src/NServiceBus.NHibernate.TransactionalSession/NHibernateTransactionalSessionOptionsCustomization.cs:7]: I wanted to make it explicit what is exactly (in terms of settings) that we need in the extension.

File: src/NServiceBus.NHibernate.TransactionalSession.AcceptanceTests/EndpointConfigurer.cs (hunk 1 above)
Comment (tmasternak) [reply src/NServiceBus.NHibernate.TransactionalSession.AcceptanceTests/EndpointConfigurer.cs:6]: This class was taken `as-is` from the main package's atts. I'm pretty sure `abstract` is not needed but wanted to make sure the code is as close to each other as possible.

File: src/NServiceBus.NHibernate.TransactionalSession.Tests/ApiApprovals.cs [hunk 6]
```diff
@@ -0,0 +1,21 @@
+﻿namespace NServiceBus.Persistence.CosmosDB.Tests
```
Comment (timbussmann) [inline src/NServiceBus.NHibernate.TransactionalSession.Tests/ApiApprovals.cs:1]: hehe, I ran into the same issue 😄

File: src/NServiceBus.NHibernate.TransactionalSession.AcceptanceTests/TransactionSessionDefaultServer.cs [hunk 7]
```diff
@@ -0,0 +1,25 @@
...
+
+    public class TransactionSessionDefaultServer : DefaultServer
+    {
+        public override Task<EndpointConfiguration> GetConfiguration(RunDescriptor runDescriptor, EndpointCustomizationConfiguration endpointConfiguration,
+            Func<EndpointConfiguration, Task> configurationBuilderCustomization) =>
+            base.GetConfiguration(runDescriptor, endpointConfiguration, async configuration =>
+            {
+                // Explicitly enforcing the type to be scanned. Otherwise the scanner would not pick it up in the acceptance tests
+                endpointConfiguration.TypesToInclude.Add(typeof(TransactionalSession));
+                endpointConfiguration.TypesToInclude.Add(typeof(NHibernateTransactionalSession));
```
Comment (timbussmann) [inline src/NServiceBus.NHibernate.TransactionalSession.AcceptanceTests/TransactionSessionDefaultServer.cs:17]: I still think this shouldn't be required. If this is required with the latest version of NSB, I think we need to smoke test the actual behavior when disabling file scanning on a real endpoint.

**Comment by danielmarbach** [comment]
Let's align this PR with the changes we have done do AzureTable, Cosmos, Mongo and Raven

**Comment by tmasternak** [comment]
LGTM

**Pull request merged**